com.unity.netcode.gameobjects/Components/HalfVector3.cs

```diff
@@ -0,0 +1,114 @@
+using System.Runtime.CompilerServices;
+using UnityEngine;
+
+namespace Unity.Netcode.Components
+{
+    /// <summary>
+    /// Half float precision <see cref="Vector3"/>
+    /// </summary>
+    public struct HalfVector3 : INetworkSerializable
+    {
+        /// <summary>
+        /// The half float precision value of the x-axis as a <see cref="ushort"/>
+        /// </summary>
+        public ushort X => Axis[0];
+        /// <summary>
+        /// The half float precision value of the y-axis as a <see cref="ushort"/>
+        /// </summary>
+        public ushort Y => Axis[1];
+        /// <summary>
+        /// The half float precision value of the z-axis as a <see cref="ushort"/>
+        /// </summary>
+        public ushort Z => Axis[2];
+
+        public ushort[] Axis;
+
+        private HalfVector3AxisToSynchronize m_HalfVector3AxisToSynchronize;
+
+        /// <summary>
+        /// The serialization implementation of <see cref="INetworkSerializable"/>
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        /// <param name="serializer"></param>
+        public void NetworkSerialize<T>(BufferSerializer<T> serializer) where T : IReaderWriter
+        {
+            for (int i = 0; i < 3; i++)
+            {
+                if (m_HalfVector3AxisToSynchronize.SyncAxis[i])
+                {
+                    var axisValue = Axis[i];
+                    serializer.SerializeValue(ref axisValue);
```

Comment:
You could get even more optimization out of this by using `BytePacker.WriteValuePacked(serializer.GetWriter(), Axis[i])` and `ByteUnpacker.ReadValuePacked(serializer.GetWriter, out Axis[i])`

Reply:
Would like to but the ushort values are half precision floats which are already as packed as they can get.